src/Metadata/Api/DataProvider.php

```diff
@@ -52,6 +52,25 @@ public function providedData(string $className, string $methodName): ?array
         }
 
         if ($dataProvider->isNotEmpty()) {
+            if ($testWith->isNotEmpty()) {
+                $method = new ReflectionMethod($className, $methodName);
+
+                Event\Facade::emitter()->testTriggeredPhpunitWarning(
+                    new TestMethod(
+                        $className,
+                        $methodName,
+                        $method->getFileName(),
+                        $method->getStartLine(),
+                        Event\Code\TestDoxBuilder::fromClassNameAndMethodName(
+                            $className,
+                            $methodName,
+                        ),
+                        MetadataCollection::fromArray([]),
+                        Event\TestData\TestDataCollection::fromArray([]),
+                    ),
+                    'Mixing #[DataProvider] and #[TestWith] attributes is not supported, only the data provided by #[DataProvider] will be used',
```

Comment:
this will be confusing when using `DataProviderExternal` or `TestWithJson`. The message should probably deal better with such cases

Reply:
good point, didn't think about all these other types of providers. will adjust the wording.